### PR TITLE
VC-4214 add save set kinds

### PR DESCRIPTION
**Pull request by alex-bodnar**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: database/vipcoin/chain/accounts/set-kinds.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import (
+	"context"
+	"database/sql"
+
+	accountstypes "git.ooo.ua/vipcoin/chain/x/accounts/types"
+	"git.ooo.ua/vipcoin/lib/filter"
+	"github.com/forbole/bdjuno/v2/database/types"
+)
+
+func (r Repository) SaveSetKinds(msg ...*accountstypes.MsgSetKinds) error {
```
Comment (nikkibuster) [inline database/vipcoin/chain/accounts/set-kinds.go]: add comment please

File: database/vipcoin/chain/accounts/set-kinds.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	if err := tx.Commit(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r Repository) GetSetKinds(accfilter filter.Filter) ([]*accountstypes.MsgSetKinds, error) {
```
Comment (nikkibuster) [inline database/vipcoin/chain/accounts/set-kinds.go]: same

File: database/vipcoin/chain/accounts/set-kinds.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	if len(msg) == 0 {
+		return nil
+	}
+
+	tx, err := r.db.BeginTxx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return err
+	}
+
+	defer tx.Rollback()
```
Comment (nikkibuster) [inline database/vipcoin/chain/accounts/set-kinds.go]: defer func() {
    _ = tx.Rollbacl()
}()

File: database/vipcoin/chain/accounts/set-kinds.go (hunk 2 above)
Comment (nikkibuster) [reply database/vipcoin/chain/accounts/set-kinds.go]: GetSetKinds => GetKinds

File: database/vipcoin/chain/accounts/set-kinds.go (hunk 1 above)
Comment (nikkibuster) [reply database/vipcoin/chain/accounts/set-kinds.go]: SaveSetKinds => SaveKinds

File: database/vipcoin/chain/accounts/set-kinds.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
+/*
```
Comment (nikkibuster) [inline database/vipcoin/chain/accounts/set-kinds.go]: rename file to set_kinds.go

File: _tests/database/accounts/set-kinds_test.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
+/*
```
Comment (nikkibuster) [inline _tests/database/accounts/set-kinds_test.go]: rename file

File: modules/vipcoin/chain/accounts/fields.go [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+ * Copyright 2022 Business Process Technologies. All rights reserved.
+ */
+
+package accounts
+
+// Defines fields.
+const (
+	FieldID   = "id"
+	FieldHash = "hash"
+)
```
Comment (nikkibuster) [inline modules/vipcoin/chain/accounts/fields.go:11]: add new line please

File: volume/config.tmpl.yaml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    host: localhost
+    port: 5432
+    user: postgres
+    password: postgres
+    schema: public
+    max_open_connections: 10
+    max_idle_connections: 10
+logging:
+    level: debug
+    format: text
```
Comment (nikkibuster) [inline volume/config.tmpl.yaml:41]: add new line please

File: _tests/database/accounts/set-kinds_test.go [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Repository.GetSetKinds() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Repository.GetSetKinds() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
```
Comment (nikkibuster) [inline _tests/database/accounts/set-kinds_test.go]: add new line please

File: _tests/database/accounts/set-kinds_test.go [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+	accountstypes "git.ooo.ua/vipcoin/chain/x/accounts/types"
+	"git.ooo.ua/vipcoin/lib/filter"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	accountsdb "github.com/forbole/bdjuno/v2/database/vipcoin/chain/accounts"
+	_ "github.com/jackc/pgx/stdlib"
+	"github.com/jmoiron/sqlx"
+)
+
+func TestRepository_SaveSetKinds(t *testing.T) {
+	// db, err := sqlx.Connect("pgx", "host=10.10.1.79 port=5432 user=postgres dbname=juno password=postgres sslmode=disable")
```
Comment (nikkibuster) [inline _tests/database/accounts/set-kinds_test.go]: do we need this unused code line?

File: database/vipcoin/chain/accounts/set-kinds.go (hunk 3 above)
Comment (alex-bodnar) [reply database/vipcoin/chain/accounts/set-kinds.go]: А зачем так делать? tx.Rollback() и так функция, и выполниться при выходе.

Comment (vasylNaumenko) [reply database/vipcoin/chain/accounts/set-kinds.go]: warning о том что не обрабатывается ошибка

**Comment by vasylNaumenko** [comment]
LGTM

File: database/vipcoin/chain/accounts/set_kinds.go [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+			VALUES 
+			(:creator, :hash, :kinds)`
+
+	for _, kinds := range msg {
+		if _, err := tx.NamedExec(query, toSetKindsDatabase(kinds)); err != nil {
+			return err
+		}
+	}
+
+	if err := tx.Commit(); err != nil {
```
Comment (nikkibuster) [inline database/vipcoin/chain/accounts/set_kinds.go]: return tx.Commit()

**Review by nikkibuster — APPROVED** [review]
LGTM

**Pull request merged**